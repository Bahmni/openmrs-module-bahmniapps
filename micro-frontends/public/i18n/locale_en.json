{
  "NO_FORM": "No Form found for this patient....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Observation Forms",
  "LOADING_MESSAGE": "Loading... Please Wait",
  "NO_ALLERGIES": "No Allergies recorded for this patient.",
  "BACK_TO_ALLERGEN": "Back to Allergies",
  "ALLERGIES_HEADING": "Allergies and Reactions",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergies",
  "SEVERITY": "Severity",
  "SAVE": "Save",
  "CANCEL": "Cancel",
  "NO_ALLERGENS_FOUND": "No Allergen found",
  "SEARCH_ALLERGEN": "Search Allergen",
  "REACTIONS": "Reaction(s)",
  "SEARCH_REACTION": "Search Reaction",
  "COMMON_REACTIONS": "Common Reactions",
  "ALLERGEN": "Allergen",
  "EDIT_FORM_ERROR_MESSAGE": "Please enter a value in the mandatory fields or correct the value in the highlighted fields to proceed",
  "EMPTY_NOTES_ERROR": "Note cannot be empty",
  "OT_NOTES": "Notes",
  "FROM": "From",
  "TO": "To",
  "FROM_DATE_BEFORE_TO_DATE_ERROR": "From date should be before To date",
  "DATE_OUT_OF_RANGE_ERROR": "Please select date within the valid range",
  "DELETE_NOTE_TITLE": "Delete Note",
  "YES_KEY": "Yes",
  "NO_KEY": "No",
  "DELETE_NOTE_MESSAGE": "Are you sure you want to delete this note?",
  "NOTES_PLACEHOLDER": "Enter a maximum of 150 characters",
  "UPDATE_NOTE_TITLE": "Update Note",
  "ADD_NOTE_TITLE": "Add Note",
  "LOADING": "Loading...",
  "UPDATE": "Update",
  "AKNOWLEDGE_BUTTON": "Acknowledge",
  "AKNOWLEDGEMENT_REQUIRED_TEXT": "Acknowledgement required",
  "NO_DRUGS_TO_BE_AKNOWLEDGED_TEXT": "You have no new drugs to be acknowledged"
}