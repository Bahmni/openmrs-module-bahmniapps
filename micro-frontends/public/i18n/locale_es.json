{
  "NO_FORM": "No se encontró ningún formulario para este paciente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formularios de observación",
  "LOADING_MESSAGE": "Cargando... Espere por favor",
  "NO_ALLERGIES": "No se registraron alergias para este paciente.",
  "BACK_TO_ALLERGEN": "Volver a Alergias",
  "ALLERGIES_HEADING": "Alergias y reacciones",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Alergias",
  "SEVERITY": "Gravedad",
  "SAVE": "Ahorrar",
  "CANCEL": "Cancelar",
  "NO_ALLERGENS_FOUND": "No se encontró ningún alérgeno",
  "SEARCH_ALLERGEN": "Buscar alérgeno",
  "REACTIONS": "Reacciones",
  "SEARCH_REACTION": "Reacción de búsqueda",
  "COMMON_REACTIONS": "Reacciones comunes",
  "ALLERGEN": "Alérgeno",
  "EDIT_FORM_ERROR_MESSAGE": "Ingrese un valor en los campos obligatorios o corrija el valor en los campos resaltados para continuar"
}
