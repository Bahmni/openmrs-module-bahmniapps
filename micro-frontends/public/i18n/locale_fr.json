{
  "NO_FORM": "Aucun formulaire trouvé pour ce patient....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formulaires d'observations",
  "LOADING_MESSAGE": "Chargement... Veuillez patienter",
  "NO_ALLERGIES": "Aucune allergie enregistrée pour ce patient.",
  "BACK_TO_ALLERGEN": "Retour à Allergies",
  "ALLERGIES_HEADING": "Allergies et réactions",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergies",
  "SEVERITY": "Gravité",
  "SAVE": "Sauvegarder",
  "CANCEL": "Annuler",
  "NO_ALLERGENS_FOUND": "Aucun allergène trouvé",
  "SEARCH_ALLERGEN": "Rechercher un allergène",
  "REACTIONS": "Réaction(s)",
  "SEARCH_REACTION": "Réaction à la recherche",
  "COMMON_REACTIONS": "Réactions courantes",
  "ALLERGEN": "Allergène",
  "EDIT_FORM_ERROR_MESSAGE": "Veuillez saisir une valeur dans les champs obligatoires ou corriger la valeur dans les champs en surbrillance pour continuer",
  "ACKNOWLEDGEMENT": "Reconnaître",
  "ENTER_NOTES": "Saisir des notes",
  "NOTE": "Note",
  "ALLERGY_SAVED_SUCCESS": "Allergie enregistrée avec succès",
  "ERROR_SAVING_ALLERGY": "Erreur lors de l'enregistrement de l'allergie",
  "ACKNOWLEDGEMENT_REQUIRED_TEXT": "Accusé de réception requis",
  "MALE": "Homme",
  "FEMALE": "Femme",
  "OTHER": "Autre",
  "ADDITIONAL_COMMENT_ALLERGY": "Commentaires supplémentaires tels que la date d'apparition, etc.",
  "ALLERGEN_SEARCH_PLACEHOLDER": "Saisissez pour rechercher un allergène"
}
