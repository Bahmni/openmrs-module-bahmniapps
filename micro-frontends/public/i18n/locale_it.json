{
  "NO_FORM": "Nessun modulo trovato per questo paziente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Moduli di osservazione",
  "LOADING_MESSAGE": "Caricamento in corso... Attendere prego",
  "NO_ALLERGIES": "Nessuna allergia registrata per questo paziente.",
  "BACK_TO_ALLERGEN": "Torniamo alle allergie",
  "ALLERGIES_HEADING": "Allergie e reazioni",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergie",
  "SEVERITY": "Gravit√†",
  "SAVE": "Salva",
  "CANCEL": "Cancellare",
  "NO_ALLERGENS_FOUND": "Nessun allergene trovato",
  "SEARCH_ALLERGEN": "Cerca allergene",
  "REACTIONS": "Reazione(i)",
  "SEARCH_REACTION": "Cerca reazione",
  "COMMON_REACTIONS": "Reazioni comuni",
  "ALLERGEN": "Allergene",
  "EDIT_FORM_ERROR_MESSAGE": "Inserisci un valore nei campi obbligatori o correggi il valore nei campi evidenziati per procedere"
}
