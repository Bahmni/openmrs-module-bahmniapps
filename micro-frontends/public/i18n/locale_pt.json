{
  "NO_FORM": "Nenhum formulário encontrado para este paciente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formulários de Observação",
  "LOADING_MESSAGE": "Carregando... Aguarde",
  "NO_ALLERGIES": "Nenhuma alergia registrada para este paciente.",
  "BACK_TO_ALLERGEN": "Voltar para Alergias",
  "ALLERGIES_HEADING": "Alergias e reações",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Alergias",
  "SEVERITY": "Gravidade",
  "SAVE": "Salvar",
  "CANCEL": "Cancelar",
  "NO_ALLERGENS_FOUND": "Nenhum alérgeno encontrado",
  "SEARCH_ALLERGEN": "Pesquisar alérgeno",
  "REACTIONS": "Reação(ões)",
  "SEARCH_REACTION": "Reação de pesquisa",
  "COMMON_REACTIONS": "Reações Comuns",
  "ALLERGEN": "Alérgeno",
  "EDIT_FORM_ERROR_MESSAGE": "Insira um valor nos campos obrigatórios ou corrija o valor nos campos destacados para prosseguir"
}
