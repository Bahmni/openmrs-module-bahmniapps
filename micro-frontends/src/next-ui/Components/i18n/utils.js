import { LS_LANG_KEY, BASE_URL } from "../../constants";

const translationsBaseUrl = "i18n";

export function getLocale() {
  return localStorage.getItem(LS_LANG_KEY) || "en";;
}

export const getTranslations = async (locale) => {
  const fileName = `locale_${locale}.json`;
  return fetchTranslations(fileName);
};

async function fetchTranslations(fileName) {
  const url = `${BASE_URL}${translationsBaseUrl}/${fileName}`;
  const response = await fetch(url);
  return response.json();
}
