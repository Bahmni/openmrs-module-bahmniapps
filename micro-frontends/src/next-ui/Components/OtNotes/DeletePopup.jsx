import React, { useState } from "react";
import PropTypes from "prop-types";
import { Modal } from "carbon-components-react";
import "../../../styles/carbon-conflict-fixes.scss";
import "../../../styles/carbon-theme.scss";
import { deleteOtNote } from "./utils";
import {FormattedMessage} from "react-intl";
import "./OtNotes.scss";
export function DeletePopup(props) {
    const { hostData, hostApi} = props;
    const [isLoading, setIsLoading] = useState(false);
    return <Modal
        open
        danger
        className="next-ui ot-notes-popup"
        modalHeading={<FormattedMessage id={"DELETE_NOTE_TITLE"} defaultMessage={"Delete Note"}/>}
        primaryButtonText={isLoading? <FormattedMessage id={"LOADING"} defaultMessage={"Loading..."}/> : <FormattedMessage id={"YES"} defaultMessage={"Yes"}/>}
        secondaryButtonText={<FormattedMessage id={"NO"} defaultMessage={"No"}/>}
        onRequestClose={hostApi?.onClose}
        onSecondarySubmit={hostApi?.onClose}
        onRequestSubmit={() => {
            setIsLoading(true);
            deleteOtNote(hostData?.noteId).then(() => {
                setIsLoading(false);
                hostApi?.onSuccess();
            });
        }}
    >
        <FormattedMessage id={"DELETE_NOTE_MESSAGE"} defaultMessage={"Are you sure you want to delete this OT Note?"}/>
    </Modal>
}

DeletePopup.propTypes = {
    hostData: PropTypes.Object,
    hostApi: PropTypes.Object,
}
