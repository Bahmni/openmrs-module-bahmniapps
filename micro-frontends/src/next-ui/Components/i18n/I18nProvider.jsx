import PropTypes from "prop-types";
import React, { useEffect, useMemo, useState } from "react";
import { IntlProvider } from "react-intl";
import { getLocale, getTranslations } from "./utils";

export function I18nProvider({ children }) {
  const [messages, setMessages] = useState(undefined);
  const locale = useMemo(getLocale, []);

  useEffect(() => {
    getTranslations(locale).then(setMessages);
  }, []);

  if (!messages) {
    return <div></div>;
  }

  return (
    <IntlProvider defaultLocale="en" locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
}

I18nProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
