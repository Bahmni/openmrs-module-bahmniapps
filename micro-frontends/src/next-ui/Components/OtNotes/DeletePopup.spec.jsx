import React from "react";
import {render, screen, waitFor} from "@testing-library/react";
import { DeletePopup } from "./DeletePopup";
import {I18nProvider} from "../i18n/I18nProvider";

const mockDeletePopup = jest.fn();

jest.mock('../i18n/utils');
jest.mock("./utils", () => ({
    deleteOtNote : () => mockDeletePopup(),
}));
describe("DeletePopup", () => {
    it("should render", async () => {
        const {container, getByText} = render(<I18nProvider><DeletePopup hostData={{noteId: 10}}/></I18nProvider>);
        await waitFor(() => {
            expect(getByText("Delete Note")).toBeTruthy();
            expect(container).toMatchSnapshot();
        });
    });
    it("should call deleteOtNote on submit", async () => {
        const {getByText}= render(<I18nProvider><DeletePopup hostData={{noteId: 10}}/></I18nProvider>);
        await waitFor(() => {
            expect(getByText("Delete Note")).toBeTruthy();
        });
        mockDeletePopup.mockResolvedValue(() => {})
        screen.getByText("Yes").click();
        expect(mockDeletePopup).toHaveBeenCalled();
    });
});