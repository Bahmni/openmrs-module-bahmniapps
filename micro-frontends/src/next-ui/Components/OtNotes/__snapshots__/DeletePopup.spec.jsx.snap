// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePopup should render 1`] = `
<div>
  <div
    class="bx--modal bx--modal-tall is-visible bx--modal--danger next-ui ot-notes-popup"
    role="presentation"
  >
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
    <div
      aria-label="[object Object]"
      aria-modal="true"
      class="bx--modal-container"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="bx--modal-header"
      >
        
        <h3
          class="bx--modal-header__heading"
          id="bx--modal-header__heading--modal-1"
        >
          Delete Note
        </h3>
        <button
          aria-label="close"
          class="bx--modal-close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--modal-close__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-labelledby="bx--modal-header__heading--modal-1"
        class="bx--modal-content"
        id="bx--modal-body--modal-1"
      >
        Are you sure you want to delete this note?
      </div>
      <div
        class="bx--modal-footer bx--btn-set"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          No
        </button>
        <button
          aria-describedby="danger-description-2"
          class="bx--btn bx--btn--danger"
          tabindex="0"
          type="button"
        >
          <span
            class="bx--visually-hidden"
            id="danger-description-2"
          >
            danger
          </span>
          Yes
        </button>
      </div>
    </div>
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
  </div>
</div>
`;
